src/installer/corehost/cli/hostpolicy/args.cpp

```diff
@@ -98,10 +98,12 @@ bool parse_arguments(
         init.additional_deps_serialized,
         init.deps_file,
         init.probe_paths,
+        /* init_for_component */ false,
```

Comment:
Nit: I know `component` is horribly overloaded in our world. My initial assumption for `init_for_component` was that it referred to the app/component split, so it meant both managed plugins and native host components, but it was false here for the `libhost` / native host component scenario. Maybe something more specific to its use for component dependency resolution?

(I guess in this usage it doesn't make much of a difference, since we won't be setting the app root when initializing for `libhost` / native host component).

Reply:
Actually, what is the expectation for things like loading a .NET 5 COM component? I recall (a long time ago) talking about it being a reasonable thing to support, but it looks like that is not the case from the code.

Reply:
I'll rename it - what about `init_from_file_system` (very specific)... it could be `init_for_plugin` but we don't use "plugin" anywhere else...

I don't understand the COM component comment - if you mean the discussion to support single-file plugins... this is not it. This is solely the scenario where the app is a single-file and it uses ADR to load a plugin.

Reply:
Sorry, it was a bit of an aside - I had thought of COM components because of my initial interpretation of `init_for_component`. I was talking about a single-file app loading a (normal, non-single-file) .NET 5.0 COM component.

Reply:
Didn't try - but it SHOULD work :wink: - I can't think of a reason why it would not work - `hostfxr` is actually mostly untouched by single-file and most of the magic needed to do native hosting APIs is in `hostfxr`. 
Unfortunately we haven't tried lot of things 😞

Reply:
Heh, okay. I was thinking it would hit the problem where `comhost` doesn't know about single-file and we'd load a separate `hostfxr`/`hostpolicy`/runtime (purely from a glance the code - definitely haven't tried it).

Reply:
Crap - you're right... I didn't even think about that... yeah - that won't work. I'll try, but I'm pretty sure you're right.

Reply:
I filed https://github.com/dotnet/runtime/issues/40802#issuecomment-673728616 to track the fact that .NET Core COM server inside single-file COM client will be weird.